common/src/main/java/com/microsoft/identity/common/internal/servertelemetry/RequestTelemetry.java

```diff
@@ -0,0 +1,156 @@
+        }
+
+        final String methodName = ":putTelemetry";
+        final String schemaCompliantValueString = Schema.getSchemaCompliantString(value);
+
+        if (Schema.isCommonField(key, mIsCurrentRequest)) {
+            putInCommonTelemetry(key, schemaCompliantValueString);
+        } else if (Schema.isPlatformField(key, mIsCurrentRequest)) {
+            putInPlatformTelemetry(key, schemaCompliantValueString);
+        } else {
+            Logger.verbose(
+                    TAG + methodName,
+                    "Supplied key not added to Server telemetry map " +
+                            "as it is not part of either common or platform schema."
+            );
+        }
+    }
+
+    String getSchemaVersion() {
+        return mSchemaVersion;
+    }
+
+    Map<String, String> getCommonTelemetry() {
+        return mCommonTelemetry;
+    }
+
+    Map<String, String> getPlatformTelemetry() {
+        return mPlatformTelemetry;
+    }
+
+    String getCompleteTelemetryHeaderString() {
+        final String methodName = ":getCompleteTelemetryHeaderString";
+
+        if (StringUtil.isEmpty(mSchemaVersion)) {
+            Logger.verbose(
+                    TAG + methodName,
+                    "SCHEMA_VERSION is null or empty. " +
+                            "Telemetry Header String cannot be formed."
+            );
+
+            return null;
+        }
+
+        final String schemaVersionString = Schema.getSchemaCompliantString(mSchemaVersion);
+        final String commonSchemaString = getCommonTelemetryHeaderString();
+        final String platformSchemaString = getPlatformTelemetryHeaderString();
+        return schemaVersionString + "|" + commonSchemaString + "|" + platformSchemaString;
+    }
+
+    private String getCommonTelemetryHeaderString() {
+        final String[] commonFields = Schema.getCommonFields(mIsCurrentRequest);
+        return getTelemetryHeaderStringFromFields(commonFields, mCommonTelemetry);
+    }
+
+    private String getPlatformTelemetryHeaderString() {
+        final String[] platformFields = Schema.getPlatformFields(mIsCurrentRequest);
+        return getTelemetryHeaderStringFromFields(platformFields, mPlatformTelemetry);
+    }
+
+    /**
+     * This method loops over provided telemetry fields and creates a header string for those fields.
+     * It is important to ensure that the fields array passed to this method is in the correct order,
+     * as determined in {@link Schema}.
+     * Failure to do so will return a malformed header string.
+     *
+     * @param fields    The fields that need to be included in the header string
+     * @param telemetry A HashMap of telemetry data that maps keys (fields) to their values
+     * @return a telemetry header string composed from provided telemetry fields and values
+     */
+    private String getTelemetryHeaderStringFromFields(String[] fields, Map<String, String> telemetry) {
+        if (fields == null || telemetry == null) {
```

Comment:
If you _really_ want to support `null` and `null` as params, then the parameters should be annotated as `@NonNull` (and `final`)

Reply:
Addressed